service/controller/resource/drainer/resource.go

```diff
@@ -46,3 +61,193 @@ func NewResource(config ResourceConfig) (*Resource, error) {
 func (r *Resource) Name() string {
 	return Name
 }
+
+func (r *Resource) createDrainerConfig(ctx context.Context, cl infrastructurev1alpha2.AWSCluster, md infrastructurev1alpha2.AWSMachineDeployment, instanceID, privateDNS string) error {
+	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("creating drainer config for ec2 instance %#q", instanceID))
+
+	n := md.GetNamespace()
+	c := &g8sv1alpha1.DrainerConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Annotations: map[string]string{
+				annotation.InstanceID: instanceID,
+			},
+			Labels: map[string]string{
+				label.Cluster:           key.ClusterID(&md),
+				label.MachineDeployment: key.MachineDeploymentID(&md),
```

Comment:
I added the label here because I saw Node Pool reconciliation draining nodes of other Node Pools.